src/frontend/app/View/codeView/codeEditViews/CodePipeView.js

```diff
@@ -154,4 +155,29 @@ export default class CodePipeView extends CodeEditView {
       return true;
     });
   }
-}
+
+  //delete a pipe
+  deletePipe() {
+    let cur = this;
+    let attributeObjectRegex = '<[\\S]*?[^"/][pP]ipe[\\s\\t\\n][^]*?>[^]*?<[/][\\S]*?[^"/][pP]ipe>';
+    let matches = this.editor.getModel().findMatches(attributeObjectRegex, false, true, false, false);
+    let name = localStorage.getItem("activePipe");
+    matches.forEach(function(item, index) {
+      let pipe = cur.editor.getModel().getValueInRange(item.range);
+      if (pipe.split(/[\s>]/).find(word => word === 'name="' + name + '"')) {
+        let newPipe = '';
+        cur.edit(item.range, newPipe);
+      }
+    });
+  }
+
+  setEventListeners() {
```

Comment:
Don't put event listeners in the 'code' part, this feature is related with the flow.

Try to put event listeners in the controllers that is what controllers are for.